### PR TITLE
modify OAuth IMAP

### PR DESCRIPTION
**Pull request by Otijom**

### CODE REVIEW

File: source/oauthimap/index.rst [hunk 1]
```diff
@@ -1,59 +1,102 @@
-Oauth IMAP
+OAuth IMAP
 ==========
 
+.. note::
+   Microsoft is gradually removing the TLS 1.0 and 1.1 protocols for all Microsoft 365 applications. In order to keep your collector running, you need to add an **Oauth IMAP** plugin available in the marketplace.
```
Comment (cconard96) [inline source/oauthimap/index.rst:5]:
```suggestion
   Microsoft is gradually removing the TLS 1.0 and 1.1 protocols for all Microsoft 365 applications. In order to keep your collector running, you need to add the **Oauth IMAP** plugin, which is available in the marketplace.
```

File: source/oauthimap/index.rst [hunk 2]
```diff
@@ -1,59 +1,102 @@
...
-* `Microsoft (Office 365 via Azure AD) <https://docs.microsoft.com/en-us/exchange/client-developer/legacy-protocols/how-to-authenticate-an-imap-pop-smtp-application-by-using-oauth>`_.
+Here are the configuration steps including configuration phases on the Entra side.
 
-Install the Plugin
+Install the plugin
 ------------------
 
-Plugin is available inside GLPI marketplace.
+.. note::
+   This plugin is community and does not need a license. It is also available in Cloud.
```
Comment (cconard96) [inline source/oauthimap/index.rst:31]:
```suggestion
   This plugin does not require a GLPI-Network subscription. It is also available in Cloud.
```

File: source/oauthimap/index.rst [hunk 3]
```diff
@@ -1,59 +1,102 @@
...
+
+-  See you in the marketplace. Download Oauth IMAP and enable it
```
Comment (cconard96) [inline source/oauthimap/index.rst:33]:
```suggestion
-  Go to the marketplace. Download Oauth IMAP and enable it
```

File: source/oauthimap/index.rst [hunk 4]
```diff
@@ -1,59 +1,102 @@
...
+
+.. figure:: images/oauth-imap-1.png
+   :alt:
+
+-  Meet now on your `Azure tenant <https://portal.azure.com/#home>`__
+-  In the search box type **registration** then select
+   **app registrations**
```
Comment (cconard96) [inline source/oauthimap/index.rst:40]:
```suggestion
-  Open the `Azure Portal <https://portal.azure.com/#home>`__ for your tenant
-  In the search box type **registration** then select
   **App registrations**
```

File: source/oauthimap/index.rst [hunk 5]
```diff
@@ -1,59 +1,102 @@
...
+Register your Entra application
+-------------------------------
+
+Create the application
+~~~~~~~~~~~~~~~~~~~~~~
+
+-  Click on **new registration**
+-  Indicate the desired name, select the type of account supported then indicate the redirection URL (present in the configuration of the plugin from your GLPI interface:
+   https://XXXXXXXXXXXXXX/marketplace/oauthimap/front/authorization.callback.php) specifying the **Web** option
+-  Then click on **register**.
```
Comment (cconard96) [inline source/oauthimap/index.rst:54]:
```suggestion
-  Click on **New registration**
-  Enter the desired name, select the type of account supported then enter the redirection URL (present in the configuration of the plugin from your GLPI interface:
   https://XXXXXXXXXXXXXX/marketplace/oauthimap/front/authorization.callback.php) specifying the **Web** option
-  Then click on **Register**.
```

File: source/oauthimap/index.rst [hunk 6]
```diff
@@ -1,59 +1,102 @@
...
+-  Now go back to your GLPI interface **configuration > Application Aouth IMAP** and indicate the information collected previously :
 
-.. image:: images/create-authorization.png
+.. figure:: images/oauth-imap-7.png
+   :alt:
 
-You can create an authorization for each of the mailboxes you want to use in mail receivers.
-When you create an authorization for a mailbox, previously existing authorizations for this mailbox are discarded.
+-  Click **add**
+-  Now in the **Oauth authorization** tab , we will **create an authorization**
```
Comment (cconard96) [inline source/oauthimap/index.rst:89]:
```suggestion
-  Now in the **Oauth authorization** tab, click **Create an authorization**
```

File: source/oauthimap/index.rst [hunk 7]
```diff
@@ -1,59 +1,102 @@
...
 
-Configure your mail receiver
-----------------------------
+.. figure:: images/oauth-imap-8.png
+   :alt:
 
-To authenticate via Oauth on your mail receiver, you have to change the configuration of your mail receiver:
+-  When you click on :ti-add:**create authorization**, you will be redirected to the Microsoft services sign-in page
```
Comment (cconard96) [inline source/oauthimap/index.rst:94]:
```suggestion
-  When you click on :ti-add:**Create authorization**, you will be redirected to the Microsoft services sign-in page
```

**Review by cconard96 — CHANGES_REQUESTED** [review]

**Review by cconard96 — APPROVED** [review]

File: source/oauthimap/index.rst [hunk 8]
```diff
@@ -1,59 +1,102 @@
-Oauth IMAP
+OAuth IMAP
 ==========
 
+.. note::
+   Microsoft is gradually removing the TLS 1.0 and 1.1 protocols for all Microsoft 365 applications. In order to keep your collector running, you need to add the **Oauth IMAP** plugin, which is available in the marketplace.
+
+.. tip::
+   The Oauth tokens for the collector, retrieved during authentication with Azure by the OauthIMAP plugin, are “offline” tokens that delegate authorisation to a third-party application (GLPI). These tokens provide a renewal code that will be used by the application to renew them automatically without user intervention. You will therefore not be asked to re-authenticate after the 1st authorisation request (unless you change your password at a later date).
```
Comment (cedric-anne) [inline source/oauthimap/index.rst:8]: > (unless you change your password at a later date)

Are you sure of that? OAuth tokens should not be impacted by a password change.

**Review by cedric-anne — COMMENTED** [review]
This plugin also works with Google/Gmail mailboxes. You should keep a small mention of it, even if you do not produce complete documentation of the configuration process like for Azure.

File: source/OauthSSOEntra/okta.rst
Comment (trasher) [inline source/OauthSSOEntra/okta.rst:1]: This one should probably not be added.

Comment (Otijom) [reply source/OauthSSOEntra/okta.rst:1]: why do you think it shouldn't be added?

**Pull request merged**